HCL Parser doesn't support escaping double quote sign in String literals

## What is the smallest, simplest way to reproduce the problem?
```
    @Test
    void quoteEscaping() {
        rewriteRun(
          hcl(
            """
              locals {
                quotedText = "this is a double quote: \". Look at me"
              }
              """
          )
        );
    }
```

## What is the full stack trace of any errors you encountered?
```
[...]
Caused by: org.openrewrite.hcl.HclParsingException: Syntax error in file.tf at line 2:54 token recognition error at: '\n'.
	at org.openrewrite.hcl.HclParser$ForwardingErrorListener.syntaxError(HclParser.java:105)
	... 132 more
Caused by: LexerNoViableAltException('\n')
	at org.antlr.v4.runtime.atn.LexerATNSimulator.failOrAccept(LexerATNSimulator.java:309)
	at org.antlr.v4.runtime.atn.LexerATNSimulator.execATN(LexerATNSimulator.java:230)
	at org.antlr.v4.runtime.atn.LexerATNSimulator.match(LexerATNSimulator.java:114)
	at org.antlr.v4.runtime.Lexer.nextToken(Lexer.java:141)
	... 129 more
```

## Context

AFAICT this is a valid Terraform/HCL file.
https://developer.hashicorp.com/terraform/language/expressions/strings#escape-sequences says this is the proper way of escaping the quote sign.

--- a/rewrite-hcl/src/test/java/org/openrewrite/hcl/tree/HclStringTest.java
+++ b/rewrite-hcl/src/test/java/org/openrewrite/hcl/tree/HclStringTest.java
@@ -16,7 +16,6 @@
 package org.openrewrite.hcl.tree;
 
 import org.junit.jupiter.api.Test;
-import org.junitpioneer.jupiter.ExpectedToFail;
 import org.openrewrite.Issue;
 import org.openrewrite.test.RewriteTest;
 
@@ -24,7 +23,6 @@ import static org.openrewrite.hcl.Assertions.hcl;
 
 class HclStringTest implements RewriteTest {
 
-    @ExpectedToFail
     @Issue("https://github.com/openrewrite/rewrite/issues/4612")
     @Test
     void quoteEscaping() {
@@ -32,7 +30,7 @@ class HclStringTest implements RewriteTest {
           hcl(
             """
               locals {
-                quotedText = "this is a double quote: \". Look at me"
+                quotedText = "this is a double quote: \\". Look at me"
               }
               """
           )

